Sources/RedisProvider/Provider.swift

```diff
@@ -18,33 +19,66 @@ extension RedisCache: ConfigInitializable {
         guard let redis = config["redis"]?.object else {
             throw ConfigError.missingFile("redis")
         }
-
-        guard let hostname = redis["hostname"]?.string else {
-            throw ConfigError.missing(
-                key: ["hostname"],
-                file: "redis",
-                desiredType: String.self
+        
+        let encoding = redis["encoding"]?.string
+        
+        if let url = redis["url"]?.string {
+            try self.init(url: url, encoding: encoding)
+            
+            if let database = redis["database"]?.int {
+                try makeClient().command(try Command("database"), [database.description])
+            }
+        } else {
+            guard let hostname = redis["hostname"]?.string else {
+                throw ConfigError.missing(
+                    key: ["hostname"],
+                    file: "redis",
+                    desiredType: String.self
+                )
+            }
+            
+            guard let port = redis["port"]?.int?.port else {
+                throw ConfigError.missing(
+                    key: ["port"],
+                    file: "redis",
+                    desiredType: Port.self
+                )
+            }
+            
+            let password = redis["password"]?.string
+            
+            try self.init(
+                hostname: hostname,
+                port: port,
+                password: password
             )
+            
+            if let database = redis["database"]?.int {
+                try makeClient().command(try Command("database"), [database.description])
+            }
         }
-
-        guard let port = redis["port"]?.int?.port else {
-            throw ConfigError.missing(
-                key: ["port"],
-                file: "redis",
-                desiredType: Port.self
-            )
+    }
+    
+    //accepts a heroku redis connection string in the format of:
+    //redis://h:PASSWORD@URL:PORT
+    public convenience init(url: String, encoding: String?) throws {
+        let split = url.components(separatedBy: "@")
+        let secondHalf = split[1].components(separatedBy: ":")
+        
+        var splitPassword: String? = split[0].replacingOccurrences(of: "redis://h:", with: "")
```

Comment:
Yeah, I looked at that but I wasn't sure if it would act correctly with the `h:` out front. Will it? I'll try it out. I'm also still looking for a way to test it.